testsuite/openshift/objects/ingress.py

```diff
@@ -73,12 +75,25 @@ def rules(self):
         """Returns rules defined in the ingress"""
         return self.model.spec.rules
 
-    def wait_for_hosts(self, tolerate_failures: int = 5):
+    @property
+    def host(self):
+        """Returns host of the ingress"""
+        return self.model.metadata.annotations["kuadrant.dev/host.generated"]
+
+    def wait_for_ready(self):
+        """
+        Waits until the Ingress is ready,
+        temporarily it just checks for certificate as that is almost certainly created afterwards
+        """
+        return self.wait_for_certificate()
+
+    def wait_for_certificate(self, tolerate_failures: int = 5):
```

Comment:
if this should exist at all, should it be rather `_private`?

Reply:
Could be, but at the same time, I do not see the value of hiding it

Reply:
Having the public/official interface minimal